contracts/child/ChildChainManager/ChildChainManager.sol

```diff
@@ -87,8 +89,20 @@ contract ChildChainManager is
     }
 
     function _mapToken(address rootToken, address childToken) private {
+        address oldChildToken = rootToChildToken[rootToken];
+        address oldRootToken = childToRootToken[childToken];
```

Comment:
Shouldn't it be `oldChildToken`? Otherwise `oldRootToken` will be empty for the new `childToken`

Reply:
Consider these cases 

1. R1 mapped to C1. We are remapping R1 to C2.
oldChildToken will be C1. oldRootToken will be empty.
childToRootToken[C1] needs to be cleaned up in this case.
rootToChildToken[R1] will anyway be overwritten to C2.

2. R1 is mapped to C1. We are remapping R2 to C1.
oldChildToken will be empty. oldRootToken will be R1.
rootToChildToken[R1] needs to be cleaned up.
childToRootToken[C1] will anyway be overwritten to R2.

So coming to your question, yes `oldRootToken` will be empty but that is not a problem.